WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -12,13 +12,22 @@ protocol PostCoordinatorDelegate: AnyObject {
 
 class PostCoordinator: NSObject {
 
-    enum SavingError: Error, LocalizedError {
-        case mediaFailure(AbstractPost)
+    enum SavingError: Error, LocalizedError, CustomNSError {
+        case mediaFailure(AbstractPost, Error)
         case unknown
 
         var errorDescription: String? {
             Strings.genericErrorTitle
         }
+
+        var errorUserInfo: [String: Any] {
+            switch self {
+            case .mediaFailure(_, let error):
+                return [NSUnderlyingErrorKey: error]
```

Comment:
Not ideal, but it's the same approach as WordPressKit uses for generalizing the errors, so we are stuck with it.
I think we should cover this with UI automation or some sort of integration tests to avoid potential regressions in the future.